core/system/src/unvic.c

```diff
@@ -296,77 +259,73 @@ void unvic_irq_enable_all(void)
 
 void unvic_irq_pending_clr(uint32_t irqn)
 {
-    int is_irqn_registered;
-
     /* verify IRQ access privileges */
-    is_irqn_registered = unvic_acl_check(irqn);
-
-    /* enable IRQ */
-    if (is_irqn_registered) {
-        DPRINTF("IRQ %d pending status cleared\n\r", irqn);
-        NVIC_ClearPendingIRQ(irqn);
+    if (!unvic_acl_check(irqn)) {
         return;
     }
-    else {
-        HALT_ERROR(NOT_ALLOWED, "IRQ %d is unregistered; state cannot be changed", irqn);
-    }
+    g_unvic_vector[irqn].id = g_active_box;
+    DPRINTF("IRQ %d registered to box %d\n\r", irqn, g_active_box);
```

Comment:
The registration of an irqn to a box is something we do often in these functions. Consider placing this into an itty bitty static function, so we don't have to repeat the code and DPRINTF string so much.

Heck, even the call and check of `unvic_acl_check` is done very often. Maybe a new function like `unvic_irq_register` would be good. You'd use it like so.

```C
/* Attempt to register for the IRQ. */
if (!unvic_irq_register(irqn)) {
    /* We couldn't claim the interrupt. */
    return 0;
}
```

This is just a suggestion. Do what you think makes sense.